pychromecast/controllers/youtube.py

```diff
@@ -1,59 +1,252 @@
 """
 Controller to interface with the YouTube-app.
-
 Use the media controller to play, pause etc.
 """
+import re
+import threading
+
+import requests
 from . import BaseController
+from ..error import UnsupportedNamespace
 
-MESSAGE_TYPE = "type"
+YOUTUBE_BASE_URL = "https://www.youtube.com/"
+BIND_URL = YOUTUBE_BASE_URL + "api/lounge/bc/bind"
+LOUNGE_TOKEN_URL = YOUTUBE_BASE_URL +\
+                   "api/lounge/pairing/get_lounge_token_batch"
+
+HEADERS = {"Origin": YOUTUBE_BASE_URL,
```

Comment:
Please extract all the logic to communicate directly with the YouTube servers into a standalone Python package that will be installed from PyPI